torchvision/datasets/utils.py

```diff
@@ -28,27 +28,24 @@
 USER_AGENT = "pytorch/vision"
 
 
-def _urlretrieve(url: str, filename: str, chunk_size: int = 1024) -> None:
-    with open(filename, "wb") as fh:
-        with urllib.request.urlopen(urllib.request.Request(url, headers={"User-Agent": USER_AGENT})) as response:
-            with tqdm(total=response.length) as pbar:
-                for chunk in iter(lambda: response.read(chunk_size), ""):
-                    if not chunk:
-                        break
-                    pbar.update(chunk_size)
-                    fh.write(chunk)
-
+def _save_response_content(
+    content: Iterator[bytes],
+    destination: str,
+    length: Optional[int] = None,
+) -> None:
+    with open(destination, "wb") as fh, tqdm(total=length) as pbar:
+        for chunk in content:
+            # filter out keep-alive new chunks
+            if not chunk:
+                continue
 
-def gen_bar_updater() -> Callable[[int, int, int], None]:
-    pbar = tqdm(total=None)
+            fh.write(chunk)
+            pbar.update(len(chunk))
 
-    def bar_update(count, block_size, total_size):
-        if pbar.total is None and total_size:
-            pbar.total = total_size
-        progress_bytes = count * block_size
-        pbar.update(progress_bytes - pbar.n)
 
-    return bar_update
```

Comment:
This was used a long time ago to update the progress bar, but hasn't been in use since. Since I touch the progress bar in this PR, I think it is ok to also remove this here. Let me know, if you disagree.

Reply:
Happy to remove, but techcnically this is public. Now that we can remove deprecated stuff with our new policy, I would suggest to go for a normal deprecation cycle.